server/src/graphql/edges/person_balance.rs

```diff
@@ -0,0 +1,45 @@
+use super::super::{nodes::Balance, PageInfo};
+use crate::db::{
+    models,
+    schema::{balance, node},
+    Pool,
+};
+use diesel::prelude::*;
+use tokio_diesel::*;
+
+#[async_graphql::SimpleObject]
+pub struct PersonBalanceEdge {
+    pub cursor: String,
+    pub node: Balance,
+}
+
+#[async_graphql::SimpleObject]
+pub struct PersonBalanceConnection {
+    pub edges: Vec<PersonBalanceEdge>,
+    pub page_info: PageInfo,
+}
+
+impl PersonBalanceConnection {
```

Comment:
Technically it's not necessary to make this an `impl` block - I did that purely as a way to make the API nicer (functions defined in impl blocks are called like methods of a class/object).

Before this change, when the Squad node needed to resolve the `members` field to a `SquadMemberConnection`, it had to query the `person_squad_connection` table itself. That works fine, but if we start having multiple things that need to query the `person_squad_connection` table then things get a bit messy - if we wanted to make a change to the schema of that table, we'd need to change all of the places that are querying the table (in fact, that's the exact situation I ran into that motivated this change)

With this change, all of the queries to `person_squad_connection` can instead be handled by methods of the `SquadMemberConnection` struct (and its reverse, `PersonSquadConnection`). So instead of writing out a diesel query to get that connection type, the nodes' resolvers can just call `SquadMemberConnection::by_squad_id(pool, id)` and everything is a lot cleaner.

I'm not sure I understand what you mean by 'directly touching'?

Reply:
You answered my question, makes sense